src/praisonai-train/praisonai_train/train/_llamacpp.py

```
@@ -0,0 +1,266 @@
+    Raises:
+        RuntimeError: with captured output if llama-cli fails.
+    """
+    binary = find_llama_binary("llama-cli")
+    prompt = prompt or "Explain multi-token prediction in one paragraph."
+    cmd = build_mtp_cmd(
+        binary, model_gguf, draft_gguf=draft_gguf,
+        spec_draft_n_max=spec_draft_n_max, server=False, ngl=ngl,
+        extra=["--prompt", prompt, "-n", str(n_predict), "--no-conversation"],
+    )
+
+    proc = subprocess.run(cmd, capture_output=True, text=True)
```

Ruff (0.15.21)

[error] `subprocess` call: check for execution of untrusted input

(S603)